### DISCUSSION TITLE
Migration path from task protocol version 1 to 2

### DISCUSSION THREAD

**Discussion by timmc-edx** [celery/celery#9400] [Q&A]
We have a service that's still using task protocol version 1 and we'd like to upgrade to version 2 and I'm trying to ensure that we have the correct runbook.

The 4.0 release notes describe [updating workers and then clients to 3.1.25 or higher](https://docs.celeryq.dev/en/stable/history/whatsnew-4.0.html#step-1-upgrade-to-celery-3-1-25). All of ours are already on celery 4 or higher. Does that mean we can safely switch the clients' celery protocol to 2 at any time, even with messages in flight? I assume that's the case, and we'll do some testing as well, but I'd like to get that confirmed first if possible. :-)

**Comment by auvipy** [comment]
most probably yes.
